src/cmd/flux-jobs.py

```diff
@@ -288,24 +319,81 @@ def format_header(fmt):
         runtime_hms="RUNTIME",
     )
 
-    def format_denumericalize(fmt):
+    def __init__(self, fmt):
         """
-        Make a format string suitable for str types only, i.e. remove
-        all floating-point and number format specifiers, but keep width
-        and justification. Used to modify user-provided format string
-        for use with headings.
+        Parse the input format fmt with string.Formatter.
+        Save off the fields and list of format tokens for later use,
+        (converting None to "" in the process)
+
+        Throws an exception if any format fields do not match the allowed
+        list of headings above.
+        """
+        from string import Formatter
+
+        self.fmt = fmt
+        #  Parse format into list of (string, field, spec, conv) tuples,
+        #   replacing any None values with empty string ""
+        l = Formatter().parse(fmt)
+        self.format_list = [[s or "" for s in t] for t in l]
```

Comment:
Question, b/c Python docs weren't clear. What is the purpose of checking for `None` and replacing with empty string?  Presumably `None` means some field was formatted incorrectly, so shouldn't that be an outright error?

Reply:
Heh, I knew I should have added a comment.

I don't know Python that well either. This is kludge so that later we don't have to deal with the `None` type when the format string is put back together.

AFAICT, `formatter.parse()` returns `None` for some of the tuple items if they don't exist in the parsed format, e.g.:

```
>>> print([l for l in f.parse("{foo} {foo:<20} prefix{foo!r:>13.2f}")])
[('', 'foo', '', None), (' ', 'foo', '<20', None), (' prefix', 'foo', '>13.2f', 'r')]
```

If there is an error parsing the format, then `formatter.parse()` throws an exception, which now is caught, e.g.:

```console
$ flux jobs -o "{foo"
flux-jobs: ERROR: Error in user format: expected '}' before end of string
```

Reply:
on this line

`self.format_list = [[s or "" for s in t] for t in l]`

aren't you checking if the tuple is equal to `None`?  Perhaps you meant to check for None on the next line? e.g.

`self.fields = [field or "" for (s, field, spec, conv) in self.format_list]`

Reply:
Actually, let me make sure I'm substituting the `conv` part of the spec back into the right place before we merge this.

Reply:
Oh, yeah, pretty sure I'm doing that completely wrong.

Reply:
> aren't you checking if the tuple is equal to None? Perhaps you meant to check for None on the next line? e.g.

Ah, I missed this comment last go-around @chu11. Let me make sure I'm doing this right. The first line is supposed iterate over the list of tuples returned from `Formatter().parse()` and first turn the list of tuples into a list of lists, and second replace any tuple contents that are `None` with an empty string, e.g
```
>>> list = [i for i in f.parse("prefix {id:>18} {username:<8.8}")]
>>> print(list)
[('prefix ', 'id', '>18', None), (' ', 'username', '<8.8', None)]
>>> list = [[ s for s in i]  for i in f.parse("prefix {id:>18} {username:<8.8}")]
>>> print (list)
[['prefix ', 'id', '>18', None], [' ', 'username', '<8.8', None]]
>>> list = [[ s or "" for s in i]  for i in f.parse("prefix {id:>18} {username:<8.8}")]
>>> print (list)
[['prefix ', 'id', '>18', ''], [' ', 'username', '<8.8', '']]
```

Finally, the `field` no longer needs to be checked for `None` since that has been replaced in the first line.